compiler/qsc/src/interpret/stateless.rs

```diff
@@ -72,28 +95,52 @@ impl Context {
         }
     }
 
+    #[must_use]
+    pub fn eval_context(&self) -> EvalContext {
+        EvalContext {
+            context: &self.context,
+            env: Env::with_empty_scope(),
+            sim: SparseSim::new(),
+            lookup: Lookup {
+                store: &self.context.store,
+            },
+            state: State::new(self.context.package),
+        }
+    }
+}
+
+impl<'a> EvalContext<'a> {
     /// # Errors
     ///
     /// Returns a vector of errors if evaluating the entry point fails.
-    pub fn eval(&self, receiver: &mut dyn Receiver) -> Result<Value, Vec<Error>> {
-        eval_expr(
-            &get_entry_expr(&self.store, self.package)?,
-            &|id| get_global(&self.store, id),
-            self.package,
-            &mut Env::with_empty_scope(),
-            &mut SparseSim::new(),
+    pub fn eval(&mut self, receiver: &mut dyn Receiver) -> Result<Value, Vec<Error>> {
```

Comment:
Maybe this method name should mention that it's starting from the entry expression, to contrast it with stepping?